index.js

```diff
@@ -70,8 +70,11 @@ export default function stringWidth(string, options = {}) {
 		}
 
 		// TODO: Use `/\p{RGI_Emoji}/v` when targeting Node.js 20.
-		if (emojiRegex().test(character)) {
-			width += 2;
+		const emojiMatch = character.match(emojiRegex());
+		if (emojiMatch) {
+			const emoji = emojiMatch[0];
+			const codePoints = [...emoji].length;
```

Comment:
I got the idea about `[...emoji].length` from

https://github.com/mathiasbynens/emoji-regex/blob/8a6871a787a9c9441ed5a341951dcdb1fb3c1d1f/README.md?plain=1#L34

Maybe there is a better/more performant approach but that looks good to me.